backend/tests/toolmodels/test_pipelines.py

```diff
@@ -0,0 +1,363 @@
+    def mock_add_user_to_repository(
+        instance: t4c_models.DatabaseT4CInstance,
+        repository_name: str,
+        username: str,
+        password: str = credentials.generate_password(),
+        is_admin: bool = False,
+    ):
+        return {}
+
+    monkeypatch.setattr(
+        t4c_repositories_interface,
+        "add_user_to_repository",
+        mock_add_user_to_repository,
+    )
+
+
+class MockOperator:
+    cronjob_counter = 0
+    triggered_cronjob_counter = 0
+    job_counter = 0
+
+    def get_cronjob_last_run_by_label(self, label_key: str, label_value: str):
+        return None
+
+    def create_cronjob(
+        self,
+        image: str,
+        environment: dict[str, str | None],
+        schedule="* * * * *",
+        timeout=18000,
+    ) -> str:
+        self.cronjob_counter += 1
+        return self._generate_id()
+
+    def trigger_cronjob(self, name: str, overwrite_environment=None) -> None:
+        self.triggered_cronjob_counter += 1
+
+    def create_job(
+        self,
+        image: str,
+        labels: dict[str, str],
+        environment: dict[str, str | None],
+        timeout: int = 18000,
+    ) -> str:
+        self.job_counter += 1
+        return self._generate_id()
+
+    def _generate_id(self) -> str:
+        return "".join(random.choices(string.ascii_lowercase, k=25))
+
+    def delete_cronjob(self, _id: str):
+        self.cronjob_counter -= 1
+        return
+
+
+@pytest.fixture(name="mockoperator")
+def fixture_mockoperator(monkeypatch: pytest.MonkeyPatch):
+    mockoperator = MockOperator()
+    monkeypatch.setattr(
+        capellacollab.sessions.operators, "OPERATOR", mockoperator
+    )
+    return mockoperator
+
+
+@pytest.mark.usefixtures("project_manager", "mockoperator", "pipeline")
+def test_get_all_pipelines_of_capellamodel(
+    project: project_models.DatabaseProject,
+    capella_model: toolmodels_models.CapellaModel,
+    client: testclient.TestClient,
+    run_nightly: bool,
+    include_commit_history: bool,
+):
+    response = client.get(
+        f"/api/v1/projects/{project.slug}/models/{capella_model.slug}/backups/pipelines"
+    )
+
+    assert response.status_code == 200
+    assert response.json() == [
+        {
+            "id": 1,
+            "k8s_cronjob_id": "unavailable",
+            "lastrun": None,
+            "t4c_model": {
+                "project_name": "default",
+                "repository_name": "test",
+                "instance_name": "default",
+            },
+            "git_model": {
+                "id": 1,
+                "name": "",
+                "path": "http://example.com",
+                "entrypoint": "test/test.aird",
+                "revision": "main",
+                "primary": True,
+                "username": "user",
+                "password": True,
+            },
+            "run_nightly": run_nightly,
+            "include_commit_history": include_commit_history,
+        }
+    ]
```

Comment:
> * the `t4c_model` has the same format as a `SimpleT4CModel` (as this was a previous error case)

Changed it, but didn't apply the `SimpleT4CModel` proposal. The pydantic validation would fail if it's not really a `SimpleT4CModel`.